transformer_engine/jax/__init__.py

```diff
@@ -2,10 +2,10 @@
 #
 # See LICENSE for license information.
 """Transformer Engine bindings for JAX"""
+from .flax import DenseGeneral, LayerNorm
```

Comment:
Hmmm, I'm thinking how to better warn the users about the deprecation rather just only have it in the documentation.

How about something like this:
```python
import warnings
def deprecate(cls, msg):
    class deprecated(cls):
        def __init__(*args, **kwargs):
            args_without_self = args[1:]
            warnings.warn(msg, DeprecationWarning)
            super(deprecated, args[0]).__init__(*args_without_self, **kwargs)
    return deprecated

import flax
DenseGeneral = deprecate(flax.DenseGeneral, "DenseGeneral is moving to transformer_engine.jax.flax module")
LayerNorm = deprecate(flax.LayerNorm, "LayerNorm is moving to transformer_engine.jax.flax module")
```
?

The `deprecate` function would obviously need to land in some utils file.

Reply:
Thoughts, @timmoon10 @ksivaman?

Reply:
Not sure what you mean by "this approach" - the one I propose? I believe with this approach it is just the e.g `jax.DenseGeneral` would show an error, since it is a different class than the `jax.flax.DenseGeneral`.

Reply:
This is a good approach. 
Extending this idea to add `deprecate_wrapper` in `transformer_engine.common.utils.py` for supporting `Class`, `Function` and `Enum`. Refer to [commits/9c8eee6](https://github.com/NVIDIA/TransformerEngine/pull/169/commits/9c8eee6b45f4bede02d9190e285a35ee11243289).